iterating-over-data/iterating-and-special-cases.md

```diff
@@ -0,0 +1,132 @@
+# Iterating and Special Cases
+
+## Learning Goals
+
+- Define the "break" keyword as a way to exit a loop
+- Define the "continue" keyword as a way to advance an iteration in a loop
+- Use the "range()" syntax combined with a for loop
+
+## Introduction
+
+There are more things we can do with loops!
+
+Are we able to stop a loop before it gets to the last element? Are we able to use `for` loops on other data structures?
+
+By learning these different pieces of syntax, we can learn to wield them wisely whenever it becomes relevant.
+
+## Vocabulary and Synonyms
+
+| Vocab      | Definition                                                       | Synonyms | How to Use in a Sentence                                                                     |
+| ---------- | ---------------------------------------------------------------- | -------- | -------------------------------------------------------------------------------------------- |
+| `break`    | A keyword that will exit an entire loop                          | -        | "We can break out of the loop using `break`," "The loop exited early because it saw `break`" |
+| `continue` | A keyword that will immediately advance one interation in a loop | -        | "The loop moved on because it saw `continue`"                                                |
+| `range()`  | A Python function that creates a sequence                        | -        | "We used a `for` loop and `range()` in order to iterate through a series of numbers.         |
+
+## `break` Exits a Loop
+
+`break` is a keyword that we can put in a `for` loop. When our code comes across `break`, it will exit the `for` loop entirely.
+
+```python
+for fruit in ["apples", "oranges", "bananas"]:
+    print("I'm in the for loop! The value of fruit is:", fruit)
+    break
+    print("This message will never print, because the for loop saw break before it saw me...")
+```
+
+We can pair this with conditionals (and other logic)!
+
+```python
+print("I'm looking for a four leaf clover in the field...")
+
+for field_item in ["grass", "grass", "more grass", "four-leaf clover", "rocks", "rocks", "more rocks"]:
+    print("...")
+    if field_item == "four-leaf clover":
+        print("I found a four-leaf clover!")
+        print("My hunt is over! I can leave now")
+        break
+    print(f"I found {field_item} in the field...")
+    print("Better keep looking.")
+```
+
+### !callout-info
+
+## for-else
+
+Turns out we can pair a `for` loop with an `else` clause! The `else` clause will run after a `for` loop finishes **if** it did **not** encounter a `break`. The `else` is on the same level of indentation as the `for`.
```

Comment:
make a callback to where `else` was first introduced.
additional explicit details about the structure of the else clause
```suggestion
We have seen `else` clauses used with `if` statements, but it turns out we can also pair them with `for` loops! The `else` clause will run after a `for` loop finishes **if** it did **not** encounter a `break`. The `else` keyword must be at the same level of indentation as the `for`, and be followed by a `:`. The `else` body begins indented one level on the following line.
```

Reply:
```suggestion
Turns out we can pair a `for` loop with an `else` clause! The `else` clause will run after a `for` loop finishes **if** it did **not** encounter a `break`. The `else` keyword must be at the same level of indentation as the `for`, and be followed by a `:`. The `else` body begins indented one level on the following line.
```

Reply:
```suggestion
It turns out we can pair a `for` loop with an `else` clause! The `else` clause will run after a `for` loop finishes **if** it did **not** encounter a `break`. The `else` keyword must be at the same level of indentation as the `for`, and be followed by a `:`. The `else` body begins indented one level on the following line.
```